Database/TestDatabasePreparator.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+/*
+ * This file is part of the Liip/FunctionalTestBundle
+ *
+ * (c) Lukas Kahwe Smith <smith@pooteeweet.org>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Liip\FunctionalTestBundle\Database;
+
+use Doctrine\Common\DataFixtures\ProxyReferenceRepository;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\DependencyInjection\Container;
+use Doctrine\Common\DataFixtures\Executor\AbstractExecutor;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Doctrine\ORM\Tools\SchemaTool;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\DBAL\Driver\PDOSqlite\Driver as SqliteDriver;
+
+/**
+ * This class is experimental!
+ */
+class TestDatabasePreparator
+{
+    const POST_FIXTURE_RESTORE = 'postFixtureRestore';
+    const POST_FIXTURE_SETUP = 'postFixtureSetup';
+
+    private $container;
+
+    /**
+     * @var ObjectManager
+     */
+    private $om;
+
+    /**
+     * @var string storage Type, e.g. 'ORM' or 'PHPRC'
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $omName;
+
+    /**
+     * @var array
+     */
+    static private $cachedMetadatas = [];
+
+    public function __construct(Container $container, ManagerRegistry $registry, $omName = null)
+    {
+        $this->container = $container;
+        $this->omName = $omName;
+        $this->om = $registry->getManager($omName);
```

Comment:
you should rather keep the registry and retrieve the object manager from it when needed, to be compatible with manager resetting